apps/btshell/src/cmd.c

```diff
@@ -103,23 +103,25 @@ parse_dev_addr(const char *prefix, const struct kv_pair *addr_types,
     if (!prefix) {
         name[0] = '\0';
     } else {
-        written = snprintf(name, sizeof(name) - 1, "%s", prefix);
+        written = snprintf(name, sizeof(name), "%s", prefix);
         if (written >= sizeof(name) || written < 0) {
             return EINVAL;
         }
     }
 
-    written = snprintf(name + written, sizeof(name) - written - 1, "%s", "addr");
-    if (written >= sizeof(name) || written < 0) {
+    rc = snprintf(name + written, sizeof(name) - written, "%s", "addr");
+    if (rc >= sizeof(name) || rc < 0) {
```

Comment:
hmm shouldn't this be   rc >= (sizeof(name) - written)) ?

(same below, but that would require extra val to hold tmp value)